okta.go

```diff
@@ -145,29 +147,40 @@ func makeRequestBody(t interface{}) io.Reader {
 }
 
 // pulls the factor we should use out of the response
-func extractTokenFactor(ores *OktaLoginResponse) (*OktaMfaFactor, error) {
+func extractTokenFactor(oktaCfg *OktaConfig, ores *OktaLoginResponse) (*OktaMfaFactor, error) {
 	factors := ores.Embedded.Factors
 	if len(factors) == 0 {
 		return nil, errors.New("MFA factors not present in response")
 	}
 
-	var tokenFactor OktaMfaFactor
+	if oktaCfg.TokenFactor != "" && oktaCfg.TokenFactor != defaultTokenFactor {
+		tokenFactor, err := findTokenFactor(factors, oktaCfg.TokenFactor)
+
+		if err == nil {
+			return tokenFactor, err
+		}
+		debugOkta("could not find preferred token factor", oktaCfg.TokenFactor)
+	}
+
+	tokenFactor, err := findTokenFactor(factors, defaultTokenFactor)
+	return tokenFactor, err
+}
+
+func findTokenFactor(factors []OktaMfaFactor, factorType string) (*OktaMfaFactor, error) {
+	var tokenFactor *OktaMfaFactor
 	for _, factor := range factors {
```

Comment:
If you want to let the user pick how they MFA, check out this other [project](https://github.com/okta/okta-openvpn/blob/b05e86437b840bdedba4fd3a20ef15ba21826beb/okta_openvpn.py#L214). It steps through all the response and makes a pretty prompt.

Reply:
Yes, it'd be nice to present a list of options the first time and let people choose which one they want... or maybe offer an `oktad configure` step that sets those things up for people.